CMakeLists.txt

```diff
@@ -35,22 +35,39 @@ add_definitions(-DBOOST_ALL_DYN_LINK)
 
 find_package(Boost REQUIRED COMPONENTS system log OPTIONAL_COMPONENTS unit_test_framework)
 find_package(libusb-1.0 REQUIRED)
-find_package(Qt5 COMPONENTS Multimedia MultimediaWidgets Bluetooth)
+find_package(Qt5 COMPONENTS Multimedia MultimediaWidgets Bluetooth Qml Quick QuickWidgets REQUIRED)
 find_package(KF5BluezQt REQUIRED)
 find_package(Protobuf REQUIRED)
 find_package(OpenSSL REQUIRED)
 find_package(rtaudio REQUIRED)
 find_package(taglib REQUIRED)
 find_package(X11 REQUIRED)
 
-if(RPI_BUILD)
+if(RPI_BUILD AND NOT GST_BUILD)
     add_definitions(-DUSE_OMX -DOMX_SKIP64BIT)
     set(BCM_HOST_LIBRARIES "/opt/vc/lib/libbcm_host.so")
     set(BCM_HOST_INCLUDE_DIRS "/opt/vc/include")
     set(ILCLIENT_INCLUDE_DIRS "/opt/vc/src/hello_pi/libs/ilclient")
     set(ILCLIENT_LIBRARIES "/opt/vc/src/hello_pi/libs/ilclient/libilclient.a;/opt/vc/lib/libvcos.so;/opt/vc/lib/libvcilcs.a;/opt/vc/lib/libvchiq_arm.so")
-endif(RPI_BUILD)
-
+endif(RPI_BUILD AND NOT GST_BUILD)
+if(GST_BUILD)
+    find_package(Qt5GStreamer)
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(GST REQUIRED gstreamer-1.0>=1.4
+    gstreamer-sdp-1.0>=1.4
+    gstreamer-video-1.0>=1.4
+    gstreamer-app-1.0>=1.4)
+    add_definitions(-DUSE_GST)
+    if(RPI_BUILD)
+        add_definitions(-DRPI)
+        findRpiRevision( RPI_REVISION )
+        if(RPI_REVISION MATCHES "a03111" OR RPI_REVISION MATCHES "b03111" OR RPI_REVISION MATCHES "a03111")
```

Comment:
According to https://www.raspberrypi.org/documentation/hardware/raspberrypi/revision-codes/README.md there are presently 5 revision codes for Raspberry Pi 4

![image](https://user-images.githubusercontent.com/2094779/82755370-0062ec80-9da1-11ea-883b-551e57575951.png)

It might be more reliable to use the last line of that file, e.g.

`Model           : Raspberry Pi 4 Model B Rev 1.1`

Note also that this is also contained in the two aliases:
```
/proc/device-tree/model
/sys/firmware/devicetree/base/model

```

Reply:
New code
![image](https://user-images.githubusercontent.com/2094779/87172252-ee320480-c2a1-11ea-8774-ee11884f683b.png)